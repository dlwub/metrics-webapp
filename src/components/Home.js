import React from 'react';
import { Link } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import { fetchQuestions } from '../redux/Questions/Questions';
import ammendUrl from './amendUrl';

export default function Home() {
  const dispatch = useDispatch();
  const handleClick = (e) => {
    const { id } = e.target;
    dispatch(fetchQuestions(ammendUrl(id)));
  };

  return (
    <>
      <div className="test-knowledge" />
      <div className="category-bar">
        <div className="category-text">QUESTIONS BY CATEGORY</div>
        <div>
          <span className="difficulty-level">Difficulty Level</span>
          <select name="difficulty-level" id="difficulty-level">
            <option value="any difficulty">Any Difficulty</option>
            <option value="easy">Easy</option>
            <option value="medium">Medium</option>
            <option value="hard">Hard</option>
          </select>
        </div>
      </div>
      <div className="categories">
        <Link className="links questions-link" to="/questions">
          <button id="general_knowledge" className="btn general_knowledge" onClick={handleClick} aria-label="general_knowledge" type="button" />
        </Link>
        <Link className="links questions-link" to="/questions">
          <button id="entertainment" className="btn entertainment" onClick={handleClick} aria-label="entertainment" type="button" />
        </Link>
        <Link className="links questions-link" to="/questions">
          <button id="science" className="btn science" onClick={handleClick} aria-label="science" type="button" />
        </Link>
        <Link className="links questions-link" to="/questions">
          <button id="social" className="btn social" onClick={handleClick} aria-label="social" type="button" />
        </Link>
      </div>
    </>
  );
}
