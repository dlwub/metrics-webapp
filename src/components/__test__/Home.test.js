import React from 'react';
import Renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import Home from '../Home';

describe('DetailsPage', () => {
  it('renders correctly', () => {
    const home = Renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(home).toMatchSnapshot();
  });
});