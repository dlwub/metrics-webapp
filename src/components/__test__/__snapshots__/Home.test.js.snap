// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsPage renders correctly 1`] = `
Array [
  <div
    className="test-knowledge"
  />,
  <div
    className="category-bar"
  >
    <div
      className="category-text"
    >
      QUESTIONS BY CATEGORY
    </div>
    <div>
      <span
        className="difficulty-level"
      >
        Difficulty Level
      </span>
      <select
        id="difficulty-level"
        name="difficulty-level"
      >
        <option
          value="any difficulty"
        >
          Any Difficulty
        </option>
        <option
          value="easy"
        >
          Easy
        </option>
        <option
          value="medium"
        >
          Medium
        </option>
        <option
          value="hard"
        >
          Hard
        </option>
      </select>
    </div>
  </div>,
  <div
    className="categories"
  >
    <a
      className="links questions-link"
      href="/questions"
      onClick={[Function]}
    >
      <button
        aria-label="general_knowledge"
        className="btn general_knowledge"
        id="general_knowledge"
        onClick={[Function]}
        type="button"
      />
    </a>
    <a
      className="links questions-link"
      href="/questions"
      onClick={[Function]}
    >
      <button
        aria-label="entertainment"
        className="btn entertainment"
        id="entertainment"
        onClick={[Function]}
        type="button"
      />
    </a>
    <a
      className="links questions-link"
      href="/questions"
      onClick={[Function]}
    >
      <button
        aria-label="science"
        className="btn science"
        id="science"
        onClick={[Function]}
        type="button"
      />
    </a>
    <a
      className="links questions-link"
      href="/questions"
      onClick={[Function]}
    >
      <button
        aria-label="social"
        className="btn social"
        id="social"
        onClick={[Function]}
        type="button"
      />
    </a>
  </div>,
]
`;
